shacl12-core/index.html

```diff
@@ -7373,6 +7373,25 @@ <h3>sh:name and sh:description</h3>
 					multiple <a>values</a>, but should only have one <a>value</a> per language tag.
 				</p>
 			</section>
+			<section id="codeIdentifier">
+				<h3>sh:codeIdentifier</h3>
+				<p>
+					Shapes may have one <a>value</a> for <code>sh:codeIdentifier</code>
+					to suggest a name that can be used for a representation of the shape in APIs, query languages
+					and similar programmatic access.
+					The values of <code>sh:codeIdentifier</code> are <a>literals</a> with <a>datatype</a> <code>xsd:string</code>
+					where the string matches the regular expression <code>^[a-zA-Z_][a-zA-Z0-9_]*$</code>.
+				</p>
+				<p>
+					There are no strict rules in SHACL Core how these values should be used.
+					At the time of writing, a typical use case would be to generate GraphQL query
+					schemas from shapes, mapping property shapes to GraphQL fields.
+					By default such schema generators could use the local name of the
+					<code>sh:path</code>, but it should use the <code>sh:codeIdentifier</code> if present.
+					In the case of complex path expressions an explicit <code>sh:codeIdentifier</code> is strongly recommended.
+					Similar requirements exist for API generators, for example in Java, JavaScript or Python.
+				</p>
```

Comment:
Suggest - feel free to ignore - have a discussion note:

```suggestion
				<p class="note">
					A typical use case would be to generate GraphQL query
					schemas from shapes, mapping property shapes to GraphQL fields.
					By default such schema generators could use the local name of the
					<code>sh:path</code>, but it should use the <code>sh:codeIdentifier</code> if present.
					In the case of complex path expressions an explicit <code>sh:codeIdentifier</code> is strongly recommended.
					Similar requirements exist for API generators, for example in Java, JavaScript or Python.
				</p>
```